### PR TITLE
feat: config option to not exit when stops leading

### PR DESCRIPTION
**Pull request by csviri**

### CODE REVIEW

**Review by adriansuarez — APPROVED** [review]
@csviri Looks good. Thanks for jumping on this. 

This enables my use case described in #2496.

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/LeaderElectionManager.java [hunk 1]
```diff
@@ -102,11 +102,15 @@ private void startLeading() {
...
-    // Note that some reconciliations might run for a very long time.
-    System.exit(1);
+    if (configurationService.getLeaderElectionConfiguration().orElseThrow().isExitOnStopLeading()) {
+      log.info("Stopped leading for identity: {}. Exiting.", identity);
+      // When leader stops leading the process ends immediately to prevent multiple reconciliations
+      // running parallel.
+      // Note that some reconciliations might run for a very long time.
+      System.exit(1);
+    } else {
+      log.info("Stopped leading, not exiting");
```
Comment (metacosm) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/LeaderElectionManager.java:112]:
```suggestion
      log.info("Stopped leading, configured not to exit");
```

Comment (metacosm) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/LeaderElectionManager.java:112]: To make it more explicit that the operator doesn't exit on purpose.

Comment (csviri) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/LeaderElectionManager.java]: makes sense, thx!

**Review by metacosm — APPROVED** [review]

**Pull request merged**